src/inmanta/db/versions/v202303010.py

```diff
@@ -0,0 +1,49 @@
+"""
+    Copyright 2023 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+from asyncpg import Connection
+
+from inmanta import data
+from inmanta.server.services.orchestrationservice import ResourceSetValidator
+
+
+async def update(connection: Connection) -> None:
+    """
+    Ensure that the configurationmodel table has the is_suitable_for_partial_compiles column and make sure
+    it's correctly populated for all existing models in the table.
+    """
+    # Create column
+    await connection.execute("ALTER TABLE public.configurationmodel ADD COLUMN is_suitable_for_partial_compiles boolean")
+    # Populate column
+    records = await connection.fetch(f"SELECT environment, version FROM {data.ConfigurationModel.table_name()}")
+    for record in records:
+        environment = record["environment"]
+        version = record["version"]
+        resources = await data.Resource.get_resources_for_version(environment, version, connection=connection)
+        resource_set_validator = ResourceSetValidator(set(resources))
```

Comment:
I think this isn't necessarily safe: it seems to me to be very similar to [ADR 2](https://github.com/inmanta/inmanta-core/blob/master/docs/adr/0002-database-upgrade-testing.md) but for the actual migration, which is more critical. I think it would be difficult to enforce this check without using these methods though, so perhaps we should just put it to `False` for all old resources and include an upgrade note in the changelog?

Reply:
It defaults to False now.